File: cpp/841.cpp
class Solution {
private:
    void dfs(vector<vector<int>>& rooms, int i, vector<bool>& trav) {
        if (trav[i])
            return;
        trav[i] = true;
        for (int j : rooms[i])
            dfs(rooms, j, trav);
    }

public:
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        vector<bool> trav(rooms.size(), false);
        dfs(rooms, 0, trav);
        for (bool b : trav)
            if (!b)
                return false;
        return true;
    }
};

File: python3/841.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        def dfs(rooms: List[List[int]], i: int, trav: List[bool]):
            if trav[i]:
                return
            trav[i] = True
            for j in rooms[i]:
                dfs(rooms, j, trav)

        trav = [False] * len(rooms)
        dfs(rooms, 0, trav)
        for i in trav:
            if not i:
                return False
        return True
